internal/validation/validation_test.go

```diff
@@ -40,31 +53,47 @@ func TestValidate(t *testing.T) {
 
 	schemas := make([]*ast.Schema, len(testData.Schemas))
 	for i, schemaStr := range testData.Schemas {
-		schemas[i] = schema.New()
-		err := schema.Parse(schemas[i], schemaStr, false)
+		s := schema.New()
+
+		s.Directives["oneOf"] = &ast.DirectiveDefinition{
+			// graphql-js includes support for @oneOf, currently in RFC
+			// This is not available in graphql-go, nor is it expected to be unless the RFC is accepted
+			// See https://github.com/graphql/graphql-js/pull/3513 & https://github.com/graphql/graphql-spec/pull/825/
+			Name:      "oneOf",
+			Desc:      "Indicates exactly one field must be supplied and this field must not be `null`.",
+			Locations: []string{"INPUT_OBJECT"},
+		}
+
+		err := schema.Parse(s, schemaStr, false)
 		if err != nil {
 			t.Fatal(err)
 		}
+		schemas[i] = s
 	}
 
 	for _, test := range testData.Tests {
 		t.Run(test.Name, func(t *testing.T) {
+			if _, ok := skip[test.Name]; ok {
+				t.Skip("Test ignored")
+			}
+
 			d, err := query.Parse(test.Query)
 			if err != nil {
-				t.Fatal(err)
+				t.Fatalf("failed to parse query: %s", err)
 			}
 			errs := validation.Validate(schemas[test.Schema], d, test.Vars, 0)
 			got := []*errors.QueryError{}
 			for _, err := range errs {
-				if err.Rule == test.Rule {
+				// graphql-js adjusted its rule naming. Preserve the existing naming to minimise changes for now
+				if rule := strings.TrimSuffix(test.Rule, "Rule"); err.Rule == rule {
 					err.Rule = ""
 					got = append(got, err)
 				}
 			}
 			sortLocations(test.Errors)
 			sortLocations(got)
 			if !reflect.DeepEqual(test.Errors, got) {
-				t.Errorf("wrong errors\nexpected: %v\ngot:      %v", test.Errors, got)
+				t.Errorf("wrong errors for rule %q\nexpected: %v\ngot:      %v", test.Rule, test.Errors, got)
```

Comment:
nitpick: this code was written long time ago and now the convention is [got before want](https://go.dev/wiki/TestComments#got-before-want).

Reply:
Will leave this alone here. I expect that its necessary to go through all the tests in the repo to apply this convention, and makes sense to do that all at once.